flutter-idea/src/io/flutter/analytics/FlutterAnalysisServerListener.java

```diff
@@ -456,52 +523,59 @@ private static boolean isDartLookupEvent(@NotNull LookupEvent event) {
   private class QuickFixListener implements DartQuickFixListener {
     @Override
     public void beforeQuickFixInvoked(@NotNull DartQuickFix intention, @NotNull Editor editor, @NotNull PsiFile file) {
-      String path = Objects.requireNonNull(file.getVirtualFile()).getPath();
-      int lineNumber = editor.getCaretModel().getLogicalPosition().line + 1;
-      @SuppressWarnings("ConstantConditions")
-      List<String> errorsOnLine =
-        pathToErrors.containsKey(path) ? pathToErrors.get(path).stream().filter(error -> error.getLocation().getStartLine() == lineNumber)
-          .map(AnalysisError::getCode).collect(Collectors.toList()) : ImmutableList.of();
-      FlutterInitializer.getAnalytics().sendEventMetric(QUICK_FIX, intention.getText(), errorsOnLine.size()); // test: quickFix()
+      maybeReport(true, (analytics) -> {
```

Comment:
rate limiting quick fixes like this is fine as I doubt users trigger a lot of quick fixes per minute.